pkg/controller/servicebindingrequest/reconciler_test.go

```diff
@@ -399,3 +399,121 @@ func TestEmptyApplicationSelector(t *testing.T) {
 	require.Equal(t, corev1.ConditionTrue, sbrOutput.Status.Conditions[0].Status)
 	require.Equal(t, 0, len(sbrOutput.Status.Applications))
 }
+
+func TestBindTwoSbrsWithSingleApplication(t *testing.T) {
+	applicationResourceRef := "applicationResourceRef"
+	matchLabels := map[string]string{
+		"connects-to": "database",
+		"environment": "testing",
+	}
+
+	backingServiceResourceRef1 := "backingServiceRef1"
+	backingServiceResourceRef2 := "backingServiceRef2"
+
+	sbrName1 := "binding-request1"
+	sbrName2 := "binding-request2"
+
+	f := mocks.NewFake(t, reconcilerNs)
+	f.AddMockedUnstructuredDeployment(applicationResourceRef, matchLabels)
+	f.AddMockedUnstructuredServiceBindingRequest(sbrName1, backingServiceResourceRef1, applicationResourceRef, deploymentsGVR, matchLabels)
+	f.AddMockedUnstructuredServiceBindingRequest(sbrName2, backingServiceResourceRef2, applicationResourceRef, deploymentsGVR, matchLabels)
+	f.AddMockedUnstructuredDatabaseCRD()
+	f.AddMockedUnstructuredDatabaseCR(backingServiceResourceRef1)
+	f.AddMockedUnstructuredDatabaseCR(backingServiceResourceRef2)
+	f.AddMockedUnstructuredSecret("db-credentials")
```

Comment:
Shouldn't there be two different secrets for two different services?

Reply:
Two different services can use the same secrete and in the end, we want to assert if sbr's name is injected in env var or not. So doesn't really matter if we are using the same secret in the unit test as long as they are consumed by two different sbrs.

Reply:
Based on the discussion in standup, one secret per SBR.
So, there should be two secrets in this case.

Reply:
I can add two secrets but it won't make any difference in the unit test, it will be impactful in e2e and real scenario as they will consume them, here we are just mentioning them to form a mock SBR, it won't affect the assertion.

Reply:
@pratikjagrut 

It's not about adding 2 secrets just for the sake of it.

Two SBRs get created. These 2 SBRs create 2 secrets. We inject secret information (not SBR name) in the application.

So, there MUST be 2 secrets present and there MUST be a check that both secrets are injected in application.

Currently, you are checking if SBR name is present in the deployment (https://github.com/redhat-developer/service-binding-operator/pull/514/files#diff-14a37dadc390fc3ec23459efc088de8eR510)

That is not correct.
 The actual check should be against, the secret name. Even if both secret name and SBR name is same, the check should still be against the secret name.

Reply:
We decided to remove the mock secret so I'm resolving this conversation.